data/tutorials/language/0it_06_hof.md

```diff
@@ -0,0 +1,930 @@
+# Higher-Order Functions
+
+**Prerequisites:**
+* Functions
+* Recursion
+
+**What you'll learn**:
+* What is a higher-order function
+* What are some common higher-order functions
+* What is function composition
+* How to compose functions
+* How to make reusable functions
+* How to make functions fit anywhere
+
+**(INTERNAL): journey:**
+* introduction – motivate the existence of higher-order functions with an example
+* how do i make new functions from existing functions?
+* composing pipelines and `|>`
+    * t-first vs t-last
+* point-free composition
+    * how we can do it
+    * why we don't do it
+* usual gotchas / patterns
+    * i need a function that does nothing: Fun.id
+    * my function is backwards: flip
+    * my function takes too many params:
+        * if you can, name it
+        * partial application in pipelines
+        * they're in the wrong order
+            * lambda wrappers (fun x -> f 123 x true)
+
+## 1. Introduction
+
+In OCaml, working with functions quickly becomes second nature. We like to think that _functions describe the world_, and like that, we usually write functions that describe how something behaves.
+
+Take for example a function that says hello to a person by name:
+
+```ocaml 
+let say_hi name = print_string ("Hello, " ^ name ^ "!\n") ;;
+```
+
+We can call this function several times, to say hello to several people:
+
+```ocaml 
+say_hi "Xavier";;
+say_hi "Sabine";;
+say_hi "Joe";;
+```
+
+If we wanted to say hello to the same person multiple times, we'd just _repeat_ the same line of code.
+
+```ocaml 
+say_hi "Camel";;
+say_hi "Camel";;
+say_hi "Camel";;
+```
+
+How would we avoid having to repeat this lines every time? One way of doing this would be to write a function to say hi 3 times:
```

Comment:
```suggestion
We can avoid having to repeat this lines every time by writing a function to say "hi" 3 times:
```
It's best to stay away from questions like this when possible. I've rephrased it into a statement, if that works for you.

Reply:
Thank you! ✨